packages/node/src/channels.ts

```diff
@@ -0,0 +1,387 @@
+import { legacy } from "@counterfactual/cf.js";
+import {
+  Address,
+  AppInstanceInfo,
+  AssetType,
+  Node
+} from "@counterfactual/common-types";
+import { ethers } from "ethers";
+
+import { IStoreService } from "./service-interfaces";
+import { zeroBalance } from "./utils";
+
+import Nonce = legacy.utils.Nonce;
+import FreeBalance = legacy.utils.FreeBalance;
+
+/**
+ * The schema of a channel is below.
+ * The following Channels class encapsulates the state and persistence of all
+ * the channels in a Node.
+ */
+
+/**
+ * The fully expanded schema for a channel:
+ * multisigAddress: {
+ *  multisigAddress: Address,
+ *  multisigOwners: Address[],
+ *  appsNonce: Nonce,
+ *  appInstances: Map<AppInstanceID,
+ *    appInstance: {
+ *      id: string,
+ *      appId: Address,
+ *      abiEncodings: {
+ *        stateEncoding: string,
+ *        actionEncoding: string
+ *      },
+ *      appState: any,
+ *      localNonce: Nonce,
+ *      dependencyNonce: Nonce,
+ *      timeout: BigNumber,
+ *      asset: {
+ *        assetType: AssetType,
+ *        token?: Address
+ *      },
+ *      deposits: Map<Address, BigNumber>
+ *    }
+ *  },
+ *  proposedAppInstances: same schema as appInstances,
+ *  freeBalances: Map<AssetType,
+ *    freeBalance: {
+ *      alice: Address,
```

Comment:
this requires checking if it breaks anything legacy-ish depending on it, but I would like to make the `{alice, bob}.*` parameters private so that people don't accidentally access them

Reply:
sgtm

This current schema is rather loosely defined tbh. Whatever schema ends up being the finalized definition that the machine understands can be reflected here.

Reply:
I see - yes, agree that this schema (due to transitive dependencies) reflects quite a bit of legacy stuff

Reply:
but it is very good that this is written out in expanded form somewhere